Add next-gen beta documentation (#1565)
slackapi/bolt-js: docs/_future/getting_started_future.md

</div>

---
### What is the next-generation platform? {#next-gen}

The next-generation platform is an exciting new way to develop apps for Slack! It is focused on 2 components:
1. **Flexible functionality**: You can now create modular, shareable, and remixable building blocks within your apps that make it easier than ever to connect and reuse functionlity. The primary components of this new functionality are [built-in](/bolt-js/future/built-in-functions) and [custom functions](/bolt-js/future/custom-functions), [workflows](/bolt-js/future/workflows), and [triggers](/bolt-js/future/triggers).
2. **Quick setup**: With our next-generation developer tools, you can quickly build secure, scalable apps. Examples of these tools are the [Slack CLI](https://api.slack.com/future/tools/cli), which allows faster development by allowing you to create and scaffold apps all from the command line. Another example are [app manifests](/bolt-js/future/app-manifest), which are configuration files for your apps that can be shared and reused to quickly set up applications.

#### Limitations

> Currently for Bolt JS, you can only create remote functions and apps which can be hosted and deployed on Heroku. Next-generation applications not built on Bolt frameworks (such as those built on [Deno](https://deno.land/)) can support non-remote functions and apps, meaning you can deploy them to secure and managed Slack infrastructure. Additionally, Bolt apps built on the next-generation platform currently do not support [Datastores](https://api.slack.com/future/datastores) that are hosted on Slack infrastructure. If you'd like to use non-remote functionality as well as Slack infrastructure Datastores, consider building your next-generation application with Deno instead! You can get started with that [here](https://api.slack.com/future/get-started).

Reviewer: @stevengill would love feedback on this to make sure I captured these differences correctly!

</div>

---
### What is the next-generation platform? {#next-gen}

The next-generation platform is an exciting new way to develop apps for Slack! It is focused on 2 components:
1. **Flexible functionality**: You can now create modular, shareable, and remixable building blocks within your apps that make it easier than ever to connect and reuse functionlity. The primary components of this new functionality are [built-in](/bolt-js/future/built-in-functions) and [custom functions](/bolt-js/future/custom-functions), [workflows](/bolt-js/future/workflows), and [triggers](/bolt-js/future/triggers).
2. **Quick setup**: With our next-generation developer tools, you can quickly build secure, scalable apps. One of the tools is the [Slack CLI](https://api.slack.com/future/tools/cli), which allows faster development by allowing you to create and scaffold apps all from the command line. Another tool are [app manifests](/bolt-js/future/app-manifest), which are configuration files for your apps that can be shared and reused to quickly set up applications.

#### Limitations

> Currently for Bolt JS, you can only create remote functions and apps which can be hosted and deployed on [Heroku](https://www.heroku.com/). Next-generation applications not built on Bolt frameworks (such as those built on [Deno](https://deno.land/)) can support non-remote functions and apps, meaning you can deploy them to secure and managed Slack infrastructure. Additionally, Bolt apps built on the next-generation platform currently do not support [Datastores](https://api.slack.com/future/datastores) that are hosted on Slack infrastructure.

Reviewer 2: Can we standardardize places that say Bolt JS and call it Bolt for Javascript instead?

If you'd like to use non-remote functionality as well as Slack infrastructure Datastores, consider building your next-generation application with Deno. You can get started with that [here](https://api.slack.com/future/get-started).

---

### Setting up the CLI {#setting-up-cli}

To build a next-generation app with Bolt JS, you'll first need to get the Slack CLI set up on your machine. You can follow the [Quickstart Guide](https://api.slack.com/future/quickstart) to get instructions on how to manually or automatically install it based on your operating system. 

Since we won't be using Deno to build or migrate our next-generation app, you can skip any instructions related to installing Deno or creating an app using a Deno template. Once you've logged into the CLI using `slack auth login` and verified your login using `slack auth list`, you can proceed with the instructions in this guide.

> 💡 When logging into a workspace using the CLI, we recommend using a workspace where you won't disrupt real work getting done — [you can create a new one for free](https://slack.com/get-started#create).

---

> ### Install dependencies {#install-dependencies}

Reviewer 2: Do you think we should add another header that covers accepting the Beta ToS?

Reviewer: ahh yes, this is a great callout! I'll add that in 🙌

---
title: Getting started
order: 0
slug: getting-started
lang: en
layout: tutorial
permalink: /future/getting-started
---

# Getting started <span class="label-beta">BETA</span>

<div class="section-content">
This guide will walk you through introducing you to the [next-generation platform](/bolt-js/future/getting-started#next-gen). It will also cover how to get started with your next-generation platform experience with Bolt for Javascript by setting up the Slack CLI and also installing the required dependencies.
</div>

---

> ### What is the next-generation platform? {#next-gen}

Reviewer 2: @ashleymcm What do you think about emphasizing that we're in beta at the moment? I know we have the beta label over the whole section, but it might be a good idea to have a sentence with some caveat language about Open Beta and how the product will change with feedback from testers. 

Looking at this from api.slack.com/future for reference. We can also include the feedback link to direct bolt users to the feedback form. 

<img width="1038" alt="Screen Shot 2022-09-14 at 9 58 04 AM" src="https://user-images.githubusercontent.com/55667998/190216601-ad45f7c8-5ed0-46bc-86f7-e18b2da08cab.png">

Reviewer 2: While we're at it... thoughts on adding extra redundancy with a specific page for Feedback that directs to the survey?

Reviewer: sounds good! I can add that in right now :D

> ⚠️ Our next generation platform is currently in beta. [Your feedback is most welcome](/bolt-js/future/feedback) - all feedback will help shape the future platform experience! 

The next-generation platform is an exciting new way to develop apps for Slack! It is focused on 2 components:
1. **Flexible functionality**: You can now create modular, shareable, and remixable building blocks within your apps that make it easier than ever to connect and reuse functionlity. The primary components of this new functionality are [built-in](/bolt-js/future/built-in-functions) and [custom functions](/bolt-js/future/custom-functions), [workflows](/bolt-js/future/workflows), and [triggers](/bolt-js/future/triggers).
2. **Quick setup**: With our next-generation developer tools, you can quickly build secure, scalable apps. One of the tools is the [Slack CLI](https://api.slack.com/future/tools/cli), which allows faster development by allowing you to create and scaffold apps all from the command line. Another tool are [app manifests](/bolt-js/future/app-manifests), which are configuration files for your apps that can be shared and reused to quickly set up applications.

#### Limitations

> Currently for Bolt JS, you can only create remote functions and apps which can be hosted on third party infrastructure (such as [Heroku](https://www.heroku.com/)). Next-generation applications not built on Bolt frameworks (such as those built on [Deno](https://deno.land/)) can support non-remote functions and apps, meaning you can deploy them to secure and managed Slack infrastructure. Additionally, Bolt apps built on the next-generation platform currently do not support [Datastores](https://api.slack.com/future/datastores) that are hosted on Slack infrastructure.

Reviewer 2: ```suggestion
Today, Bolt for Javascript supports app development using next-gen Platform features like Functions and Workflows and tools such as the Slack CLI alongside all current generally available Slack Platform features. 

We do not yet support deployment to secure and managed Slack infrastructure or Datastores API [Datastores](https://api.slack.com/future/datastores) functionality
```
@hello-ashleyintech - suggestion to make this a little more succint!

Reviewer: @srajiang should we consistently capitalize the P in platform like it is done here? (like when referring to `next generation platform`)

Reviewer 2: Oh maybe only when we add Slack to it, 

so: Slack Platform or next-generation platform?

How's that?

Reviewer: sounds good to me! ✅

Reviewer 2: @ashleymcm - double checking - did this change get made?

Reviewer: @srajiang thanks for checking in, it has been added in! 🙌  about to push up another commit that adds a newline between the two paragraphs you suggested!

Reviewer 2: Oop gotcha! I jumped the gun a little bit. Thanks!! 🙇

> If you'd like to use non-remote functionality as well as Slack infrastructure Datastores, consider building your next-generation application with Deno. You can get started with that [here](https://api.slack.com/future/get-started).

Reviewer 2: Let's get rid of the reference to remote, since devs won't really grok that unless their internal to Slack 
```suggestion
If you'd like to deploy your app with Slack infrastructure, consider building your next-generation application with the Deno Slack API. You can get started with that [here](https://api.slack.com/future/get-started).
```
